engine/src/main/java/com/sandboni/core/engine/finder/bcel/visitors/TestClassVisitor.java

```diff
@@ -55,11 +63,41 @@ public void visitMethod(Method method) {
         }
     }
 
+    private void visitAnnotations(JavaClass jc) {
+        AnnotationEntry runWithAnnotation = getAnnotation(jc.getConstantPool(), jc::getAnnotationEntries, Annotations.TEST.RUN_WITH.getDesc());
+        if (runWithAnnotation != null) {
+            String runWithValue = getAnnotationParameter(runWithAnnotation, "value");
+            // if this is a test suite - we need to save the relations on the context, for late processing in the connector
+            if(!StringUtil.isEmptyOrNull(runWithValue) && runWithValue.equals(RUN_WITH_VALUE_SUITE.getDesc())) {
+                AnnotationEntry suiteAnnotation = getAnnotation(jc.getConstantPool(), jc::getAnnotationEntries, Annotations.TEST.SUITE_CLASSES.getDesc());
+                if(suiteAnnotation != null) {
+                    this.isSuite = true;
+                    String classesList = getAnnotationParameter(suiteAnnotation, "value");
+                    List<String> testClasses = Arrays.stream(classesList.split(",")).map(s -> s.replace('/','.')).collect(Collectors.toList());
+                    String suiteClassName = jc.getClassName();
+                    // add a link from TEST_SUITE_VERTEX to suite
+                    // note: test suite vertex has to be a TestVertex in order to be able to return it as one for the results for RelatedTestsOperation.execute()..
+                    TestSuiteVertex sv = new TestSuiteVertex.Builder(suiteClassName, "", context.getCurrentLocation()).build();
+                    context.addLink(LinkFactory.createInstance(context.getApplicationId(), TEST_SUITE_VERTEX, sv, LinkType.TEST_SUITE));
+                    // add links from suite to each test class
+                    testClasses.forEach(c -> {
+                        Vertex cv = new Vertex.Builder(c, "", context.getCurrentLocation()).build();
```

Comment:
I'm not sure about having empty "actions" in the vertices. Is the action field used somewhere else that requires to be empty? If not, can you try to define meaningful actions like "run" or something that describes this vertex?

Reply:
We will need to handle this is the maven-plugin. we'll see if we need to put a dummy value but the end result should be to send the surefire only the class name.

Reply:
FYI I took a look at the sandboni-maven plugin. it looks like it knows how to handle empty actions. in com/jpmorgan/cib/core/sandboni/maven/utils/TestParser.java:42 (parseMap method) I see:
...
            if (!CollectionUtils.isEmpty(entry.getValue())) {
                result.append("#");
                result.append(String.join("+", entry.getValue()));
            }
...

Reply:
FYI2: I took your advice and added a comment in the action field. it makes more sense.